app/controllers/users.js

```diff
@@ -32,3 +33,30 @@ exports.createUser = (req, res, next) => {
     })
     .catch(next);
 };
+
+exports.logIn = (req, res, next) => {
+  if (!req.body.email || !req.body.password) return next(errors.invalidParameters);
+
+  if (!emailIsValid(req.body.email)) return next(errors.invalidUserEmail);
+
+  let usr;
+
+  userService
+    .getUserByEmail(req.body.email)
+    .then(user => {
+      if (!user) throw errors.emailNotMatchAnyAccount;
+      usr = user;
```

Comment:
why are you doing this let bindings? If you want user to be in scope to use it down below, jus dont call then and do everything inside the same funcion.